extras/ping/ping.go

```diff
@@ -572,15 +585,47 @@ func (p *Pinger) processPacket(recv *packet) error {
 	return nil
 }
 
+func (p *Pinger) parseEchoReplyFromICMP(data []byte, from *net.IP) *Packet {
+	replyPacket, err := icmp.ParseMessage(1, data[:])
+	if err != nil {
+		log.Printf("error: %v\n", err)
+		return &Packet{}
```

Comment:
I don't like this pattern much; I'd rather return `*Packet, error` or, if the error does not matter, a `*Packet, bool` mainly because I think we should leverage the fact that Go allows returning multiple values to avoid returning special-case canary values.